scripts/tests

```diff
@@ -0,0 +1,290 @@
+            stdout=subprocess.DEVNULL,
+        )
+
+    def check(file, expect, actual):
+        if actual != expect:
+            logging.error(
+                f"Check {file} failed, expecting {expect}, but getting actual {actual}"
+            )
+            sys.exit(1)
+        else:
+            logging.info(f"Check {file} passed")
+
+    if not emulate:
+        dev = usbdev()
+
+    for scheme in ["mlkem512", "mlkem768", "mlkem1024"]:
+        file = f"elf/{scheme}-{test_type}.elf"
+        expect = expect_proc(scheme)
+        if emulate:
+            actual = actual_proc(
+                subprocess.run(
+                    ["make", "emulate run", f"ELF_FILE={file}"], capture_output=True
+                ).stdout
+            )
+        else:
+            actual = actual_proc(asyncio.run(flush_and_read(dev, platform_cfg, file)))
+
+        check(file, expect, actual)
+
+
+# cli utilities
+
+_shared_options = [
+    click.option(
+        "-cfg",
+        "--platform-cfg",
+        default="hal/stm32f4discovery.cfg",
+        type=click.Path(),
+        help="Configuration file of the specified platform",
+    ),
+    click.option(
+        "-v",
+        "--verbose",
+        is_flag=True,
+        show_default=True,
+        default=False,
+        type=bool,
+        help="Show verbose output or not",
+    ),
+    click.option(
+        "-e",
+        "--emulate",
+        is_flag=True,
+        show_default=True,
+        default=False,
+        type=bool,
+        help="Emulate on the QEMU or not",
+    ),
+]
+
+
+def add_options(options):
+    return lambda func: reduce(lambda f, o: o(f), reversed(options), func)
+
+
+# commands
+
+
+@click.command(
+    short_help="Run for the specified platform and hex file without parsing the output",
+    context_settings={"show_default": True},
+)
+@add_options(_shared_options)
+@click.option(
+    "-b",
+    "--bin",
+    type=click.Path(),
+    help="The binary hex file that you wanted to test.",
+)
+def run(platform_cfg, bin, verbose, emulate):
+    config_logger(True)
+    dev = usbdev()
+
+    try:
+        result = asyncio.run(flush_and_read(dev, platform_cfg, bin))
+        logging.info(result)
+    except asyncio.CancelledError:
+        pass
+
+
+@click.command(
+    short_help="Run functional tests", context_settings={"show_default": True}
+)
+@add_options(_shared_options)
+@click.option("-t", "--ntests", default=1, type=int, help="Number of tests")
+def func(platform_cfg, ntests, verbose, emulate):
+    try:
+        base_test(
+            "test",
+            emulate,
+            platform_cfg,
+            ntests,
+            lambda _: ntests * 3,
+            lambda output: count(output, "OK"),
+            verbose,
+        )
+    except asyncio.CancelledError:
+        pass
+
+
+@click.command(short_help="Run speed tests", context_settings={"show_default": True})
+@add_options(_shared_options)
+@click.option("-t", "--ntests", default=1, type=int, help="Number of tests")
```

Comment:
Can you please rename this one to `-i` `--iterations` to be consistent with pqm4.